### PR TITLE
SDK-482 [Node SDK] Restrict $or to string keys only

### PR DESCRIPTION
**Pull request by igor-alekseev-dev**
Pull Request Checklist
======================

Please read this for our PR Culture:
------------------------------------
https://incountry.atlassian.net/wiki/spaces/ED/pages/46858869/Pull+requests+culture+Draft

Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic:
- [SDK-482](https://incountry.atlassian.net/browse/SDK-482)

Code Quality (Before submitting PR)
------------
- [x] Ensure code compiles and passes Sonar Lint.
- [x] Do your best to ensure integration tests pass. If they don't then...
- [x] Remove, ignore, or mark pending any brittle tests.
- [x] Remove commented out code.

Target Release
--------------
- [ ] Please put the target release vehicle this needs to be bundled with.
- [ ] Did you increment the version in package.json? What is the new version?

### CODE REVIEW

File: src/validation/user-input/find-filter/find-filter.ts [hunk 1]
```diff
@@ -72,9 +72,9 @@ type FindFilterInputValue = {
   serviceKey3: FilterStringQuery;
   serviceKey4: FilterStringQuery;
   serviceKey5: FilterStringQuery;
-  createdAt: FilterStrictDateLikeQuery;
-  updatedAt: FilterStrictDateLikeQuery;
-  expiresAt: FilterDateLikeQuery;
+}
```
Comment (artemgolozubov) [inline src/validation/user-input/find-filter/find-filter.ts:75]: Please add semicolon here

**Review by RommeroInCountry — CHANGES_REQUESTED** [review]
please see comment above

File: README.md [hunk 2]
```diff
@@ -691,7 +701,7 @@ The following exact match search criteria are available:
   ]
 }
 ```
-Note: currently, nested `$or` operators are not supported
+Note: currently, nested `$or` operators and non-string fields inside $or operator are not supported.
```
Comment (RommeroInCountry) [inline README.md:704]: inside `$or` operator

**Review by RommeroInCountry — APPROVED** [review]

**Review by artemgolozubov — APPROVED** [review]
LGTM

**Pull request merged**